base.py

```diff
@@ -127,12 +127,9 @@ def limit_clause(self, select, **kw):
         if select._limit_clause is not None:
             text += '\n LIMIT ' + self.process(select._limit_clause, **kw)
         if select._offset_clause is not None:
-            text = '\n LIMIT '
             if select._limit_clause is None:
-                text += self.process(sql.literal(-1))
-            else:
-                text += '0'
-            text += ',' + self.process(select._offset_clause, **kw)
+                text += '\n LIMIT 0'
+            text += ' OFFSET '+ self.process(select._offset_clause, **kw)
```

Comment:
The `OFFSET` shouldn't be used if there's no LIMIT clause I think, as it's part of it https://clickhouse.tech/docs/en/sql-reference/statements/select/limit/